python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -594,6 +595,28 @@ def test_aggregate_e2e(
         assert row.as_pydict() == {"value": idx, "count()": 1}
 
 
+def test_plan_initial_stats(
```

Comment:
also let's add test for `execute_to_legacy_block_iterator`. A call of `ds.iter_batches()` should trigger the code path.